File: src/main/java/de/floribe2000/warships_java/direct/account/PlayersPersonalDataFull.java
package de.floribe2000.warships_java.direct.account;

import de.floribe2000.warships_java.direct.api.ErrorContainer;
import de.floribe2000.warships_java.direct.api.IApiResponse;
import de.floribe2000.warships_java.direct.api.IRequestAction;
import de.floribe2000.warships_java.direct.api.Meta;
import de.floribe2000.warships_java.direct.api.stats.OperationStatsContainer;
import de.floribe2000.warships_java.direct.api.stats.PlayerStatsContainer;
import de.floribe2000.warships_java.direct.api.typeDefinitions.Status;

import java.util.Map;

/**
 * A representation of a full data set from the api. Contains all available fields for player and
 * player pvp stats.
 */
public class PlayersPersonalDataFull implements IApiResponse {

	/**
	 * The response status from the api
	 */
	private Status status = Status.ERROR;

	/**
	 * Details about errors in case of a failed request.
	 * <p>Field is null if no errors occurred during the request!</p>
     */
    private ErrorContainer error = null;

    /**
     * The meta object of the api response
     */
    private Meta meta = null;

    private Map<String, PlayerDetails> data = null;

    public Status getStatus() {
        return this.status;
    }

    public ErrorContainer getError() {
        return this.error;
    }

    public Meta getMeta() {
        return this.meta;
    }

    public Map<String, PlayerDetails> getData() {
        return this.data;
    }

    /**
     * A representation of individual player statistics.
     */
    public static class PlayerDetails {

        private int last_battle_time = 0;

        private long account_id = 0;

        private int leveling_tier = 0;

        private int created_at = 0;

        private boolean hidden_profile = false;

        private int logout_at = 0;

        private Statistics statistics = null;

        public int getLast_battle_time() {
            return this.last_battle_time;
        }

        public long getAccount_id() {
            return this.account_id;
        }

        public int getLeveling_tier() {
            return this.leveling_tier;
        }

        public int getCreated_at() {
            return this.created_at;
        }

        public boolean isHidden_profile() {
            return this.hidden_profile;
        }

        public int getLogout_at() {
            return this.logout_at;
        }

        public Statistics getStatistics() {
            return this.statistics;
        }

        public String getNickname() {
            return this.nickname;
        }

        public long getStats_updated_at() {
            return this.stats_updated_at;
        }

        /**
         * A representation of the player's battle statistics
         */
        public static class Statistics {

            private int distance = 0;

            private int battles = 0;

			private PlayerStatsContainer pvp = null;

			private PlayerStatsContainer pve = null;

			private PlayerStatsContainer pvp_div2 = null;

			private PlayerStatsContainer pvp_div3 = null;

			private PlayerStatsContainer pve_div2 = null;

			private PlayerStatsContainer pve_div3 = null;

			private PlayerStatsContainer rank_solo = null;

            private PlayerStatsContainer rank_div2 = null;

            private PlayerStatsContainer rank_div3 = null;

            private OperationStatsContainer oper_solo = null;

            private OperationStatsContainer oper_div = null;

            private OperationStatsContainer oper_div_hard = null;

            public int getDistance() {
                return this.distance;
            }

            public int getBattles() {
                return this.battles;
            }

            public PlayerStatsContainer getPvp() {
                return this.pvp;
            }

            public PlayerStatsContainer getPve() {
                return this.pve;
            }

            public PlayerStatsContainer getPvp_div2() {
                return this.pvp_div2;
            }

            public PlayerStatsContainer getPvp_div3() {
                return this.pvp_div3;
            }

            public PlayerStatsContainer getPve_div2() {
                return this.pve_div2;
            }

            public PlayerStatsContainer getPve_div3() {
                return this.pve_div3;
            }

            public PlayerStatsContainer getRank_solo() {
                return this.rank_solo;
            }

            public PlayerStatsContainer getRank_div2() {
                return this.rank_div2;
            }

            public PlayerStatsContainer getRank_div3() {
                return this.rank_div3;
            }

            public OperationStatsContainer getOper_solo() {
                return this.oper_solo;
            }

            public OperationStatsContainer getOper_div() {
                return this.oper_div;
            }

            public OperationStatsContainer getOper_div_hard() {
                return this.oper_div_hard;
            }
		}

		private String nickname = null;

		private long stats_updated_at = 0;
	}

	@Override
	public String toString() {
		return IRequestAction.GSON.toJson(this);
	}
}

File: src/main/kotlin/de/floribe2000/warships_java/direct/account/PlayersPersonalDataFull.kt
package de.floribe2000.warships_java.direct.account

import com.google.gson.annotations.SerializedName
import de.floribe2000.warships_java.direct.api.ErrorContainer
import de.floribe2000.warships_java.direct.api.IApiResponse
import de.floribe2000.warships_java.direct.api.Meta
import de.floribe2000.warships_java.direct.api.stats.OperationStatsContainer
import de.floribe2000.warships_java.direct.api.stats.PlayerStatsContainer
import de.floribe2000.warships_java.direct.api.typeDefinitions.Status
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * A representation of a full data set from the api. Contains all available fields for player and
 * player pvp stats.
 */
@Serializable
data class PlayersPersonalDataFull(
    /**
     * The response status from the api
     */
    override val status: Status = Status.ERROR,

    /**
     * Details about errors in case of a failed request.
     *
     * Field is null if no errors occurred during the request!
     */
    override val error: ErrorContainer? = null,

    /**
     * The meta object of the api response
     */
    override val meta: Meta = Meta(),

    /**
     * A map containing the data returned by the api.
     */
    val data: Map<String, PlayerDetails?> = mapOf(),

    ) : IApiResponse {

    /**
     * A representation of individual player statistics.
     */
    @Serializable
    data class PlayerDetails(
        @SerializedName("last_battle_time")
        @SerialName("last_battle_time")
        val lastBattleTime: Long = 0,

        @SerializedName("account_id")
        @SerialName("account_id")
        val accountId: Long = 0,

        @SerializedName("leveling_tier")
        @SerialName("leveling_tier")
        val levelingTier: Int? = 0,

        @SerializedName("created_at")
        @SerialName("created_at")
        val createdAt: Long = 0,

        @SerializedName("hidden_profile")
        @SerialName("hidden_profile")
        val isHiddenProfile: Boolean = false,

        @SerializedName("logout_at")
        @SerialName("logout_at")
        val logoutAt: Long = 0,

        @SerializedName("leveling_points")
        @SerialName("leveling_points")
        val serviceRecordLevel: Int? = 0,

        val statistics: Statistics? = null,

        val nickname: String? = null,

        @SerializedName("stats_updated_at")
        @SerialName("stats_updated_at")
        val statsUpdatedAt: Long = 0,

        @SerializedName("updated_at")
        @SerialName("updated_at")
        val updatedAt: Long = 0,

        val private: PrivatePlayerData? = null,

        /**
         * The karma of the player. The api will never return a result.
         */
        private val karma: Int? = null,
    ) {
        /**
         * A representation of the player's battle statistics
         */
        @Serializable
        data class Statistics(
            val distance: Int = 0,
            val battles: Int = 0,
            val pvp: PlayerStatsContainer? = null,
            val pve: PlayerStatsContainer? = null,
            val pvp_div2: PlayerStatsContainer? = null,
            val pvp_div3: PlayerStatsContainer? = null,
            val pve_div2: PlayerStatsContainer? = null,
            val pve_div3: PlayerStatsContainer? = null,
            val rank_solo: PlayerStatsContainer? = null,
            val rank_div2: PlayerStatsContainer? = null,
            val rank_div3: PlayerStatsContainer? = null,
            val oper_solo: OperationStatsContainer? = null,
            val oper_div: OperationStatsContainer? = null,
            val oper_div_hard: OperationStatsContainer? = null,
        )
    }
}